### DISCUSSION TITLE
PLayerInfoPacket AddPlayer

### DISCUSSION THREAD

**Discussion by Jacrispys** [Minestom/Minestom#718] [Help] [answered]
Currently I have this for my packet:

`PlayerInfoPacket packet = new PlayerInfoPacket(PlayerInfoPacket.Action.ADD_PLAYER, new PlayerInfoPacket.AddPlayer(player.getUuid(), new BinaryReader()));`

but I'm not sure how I would pass the player info into the binary reader?

**Comment by KrystilizeNevaDies** [comment, marked as answer]
Is there a specific reason you do not want to use the record constructor?
https://github.com/Minestom/Minestom/blob/5579415513429b60fcdc879deec9d43e969831f2/src/main/java/net/minestom/server/network/packet/server/play/PlayerInfoPacket.java#L130-L131

**Reply by Jacrispys** [reply]
> Is there a specific reason you do not want to use the record constructor?
> 
> https://github.com/Minestom/Minestom/blob/5579415513429b60fcdc879deec9d43e969831f2/src/main/java/net/minestom/server/network/packet/server/play/PlayerInfoPacket.java#L130-L131

only because I could not find a method to get a property list from the player, otherwise that would be fine

**Reply by KrystilizeNevaDies** [reply]
So your issue is that you do not know what you want to put in the properties?
Refer to this: https://wiki.vg/Protocol#Player_Info
There is also this (minestom's handling): https://github.com/Minestom/Minestom/blob/5579415513429b60fcdc879deec9d43e969831f2/src/main/java/net/minestom/server/entity/Player.java#L1852-L1859

**Reply by Jacrispys** [reply]
I know what the properties is referring to, however getting a List of the properities from the players skin everytime seems like a lot of extra code, so I was wondering if there was a better way, perhaps I should move this and create a PR to add `Player#getProperties` ?

**Reply by KrystilizeNevaDies** [reply]
There isn't a better way atm, mostly because you aren't expected to use these packets.

If you did want to pr this, it could probably go in a utils somewhere, Player#getProperties is ambigious and could potentially be confused for attributes, statistics, player data, player nbt, e.t.c.

Though I must ask, why are you trying to construct these packets manually?

**Reply by Jacrispys** [reply]
> There isn't a better way atm, mostly because you aren't expected to use these packets.
> 
> If you did want to pr this, it could probably go in a utils somewhere, Player#getProperties is ambigious and could potentially be confused for attributes, statistics, player data, player nbt, e.t.c.
> 
> Though I must ask, why are you trying to construct these packets manually?

`Player#getSkinProperties` maybe? 

as for why, this is just the only way I know of to remove a player from the tablist.

**Reply by KrystilizeNevaDies** [reply]
To remove the player from the tablist you just need to use this:
```java
        PlayerInfoPacket packet = new PlayerInfoPacket(
                PlayerInfoPacket.Action.REMOVE_PLAYER,
                new PlayerInfoPacket.RemovePlayer(uuid));
```

**Reply by Jacrispys** [reply]
but also readd them later on...

Edit: also how would I fix the skin being broken in tab after using `ADD_PLAYER`
